yt_dlp/extractor/tumblr.py

```diff
@@ -291,21 +440,70 @@ def _real_extract(self, url):
                     video_id, headers={'Authorization': f'Bearer {self._ACCESS_TOKEN}'}, fatal=False),
                 ('response', 'timeline', 'elements', 0)) or {}
         content_json = traverse_obj(post_json, ('trail', 0, 'content'), ('content')) or []
-        video_json = next(
-            (item for item in content_json if item.get('type') == 'video'), {})
-        media_json = video_json.get('media') or {}
-        if api_only and not media_json.get('url') and not video_json.get('url'):
-            raise ExtractorError('Failed to find video data for dashboard-only post')
 
-        if not media_json.get('url') and video_json.get('url'):
-            # external video host
-            return self.url_result(
-                video_json['url'],
-                self._providers.get(video_json.get('provider'), 'Generic'))
+        # the url we're extracting from might be an original post or it might be a reblog.
+        # if it's a reblog, og:description will be the reblogger's comment, not the uploader's.
+        # content_json is always the op, so if it exists but has no text, there's no description
+        if content_json:
+            description = '\n\n'.join((
+                item.get('text') for item in content_json if item.get('type') == 'text')) or None
+        else:
+            description = self._og_search_description(webpage, default=None)
+        uploader_id = traverse_obj(post_json, 'reblogged_root_name', 'blog_name')
+
+        info_dict = {
+            'id': video_id,
+            'title': post_json.get('summary') or (blog if api_only else self._html_search_regex(
+                r'(?s)<title>(?P<title>.*?)(?: \| Tumblr)?</title>', webpage, 'title', default=blog)),
+            'description': description,
+            'uploader_id': uploader_id,
+            'uploader_url': f'https://{uploader_id}.tumblr.com/' if uploader_id else None,
+            'like_count': post_json.get('like_count'),
+            'repost_count': post_json.get('reblog_count'),
+            'age_limit': {True: 18, False: 0}.get(post_json.get('is_nsfw')),
+            'tags': post_json.get('tags'),
+        }
 
+        # for tumblr's own video hosting
+        fallback_format = None
+        formats = []
         video_url = self._og_search_video_url(webpage, default=None)
+        # for external video hosts
+        entries = []
+        ignored_providers = set()
+        unknown_providers = set()
+
+        video_jsons = (item for item in content_json if item.get('type') in ('video', 'audio'))
+        if video_jsons:
+            for video_json in video_jsons:
```

Comment:
I think we can do this, and reduce indentation below
```suggestion
        for video_json in traverse_obj(content_json, lambda _, v: v['type'] in ('video', 'audio')):
```